boards/microbit_v2/src/main.rs

```diff
@@ -0,0 +1,495 @@
+                .finalize(components::adc_component_helper!(nrf52833::adc::Adc))
+        ),
+    );
+
+    //--------------------------------------------------------------------------
+    // WIRELESS
+    //--------------------------------------------------------------------------
+
+    let ble_radio =
+        nrf52_components::BLEComponent::new(board_kernel, &base_peripherals.ble_radio, mux_alarm)
+            .finalize(());
+
+    //--------------------------------------------------------------------------
+    // LED Matrix
+    //--------------------------------------------------------------------------
+
+    let led = components::led_matrix_component_helper!(
+        nrf52833::gpio::GPIOPin,
+        nrf52::rtc::Rtc<'static>,
+        mux_alarm,
+        @fps => 60,
+        @cols => kernel::hil::gpio::ActivationMode::ActiveLow,
+            &base_peripherals.gpio_port[LED_MATRIX_COLS[0]],
+            &base_peripherals.gpio_port[LED_MATRIX_COLS[1]],
+            &base_peripherals.gpio_port[LED_MATRIX_COLS[2]],
+            &base_peripherals.gpio_port[LED_MATRIX_COLS[3]],
+            &base_peripherals.gpio_port[LED_MATRIX_COLS[4]],
+        @rows => kernel::hil::gpio::ActivationMode::ActiveHigh,
+            &base_peripherals.gpio_port[LED_MATRIX_ROWS[0]],
+            &base_peripherals.gpio_port[LED_MATRIX_ROWS[1]],
+            &base_peripherals.gpio_port[LED_MATRIX_ROWS[2]],
+            &base_peripherals.gpio_port[LED_MATRIX_ROWS[3]],
+            &base_peripherals.gpio_port[LED_MATRIX_ROWS[4]]
+
+    )
+    .finalize(components::led_matrix_component_buf!(
+        nrf52833::gpio::GPIOPin,
+        nrf52::rtc::Rtc<'static>
+    ));
+
+    //--------------------------------------------------------------------------
+    // FINAL SETUP AND BOARD BOOT
+    //--------------------------------------------------------------------------
+
+    // it seems that microbit v2 has no external clock
+    nrf52::clock::CLOCK.low_stop();
+    nrf52::clock::CLOCK.high_stop();
+    nrf52::clock::CLOCK.low_start();
+    nrf52::clock::CLOCK.high_start();
+    while !nrf52::clock::CLOCK.low_started() {}
+    while !nrf52::clock::CLOCK.high_started() {}
+
+    let platform = Platform {
+        ble_radio: ble_radio,
+        console: console,
+        gpio: gpio,
+        button: button,
+        led: led,
+        rng: rng,
+        temperature: temperature,
+        lsm303agr: lsm303agr,
+        ninedof: ninedof,
+        buzzer: buzzer,
+        adc: adc_syscall,
+        alarm: alarm,
+        ipc: kernel::ipc::IPC::new(board_kernel, &memory_allocation_capability),
+    };
+
+    let chip = static_init!(
+        nrf52833::chip::NRF52<Nrf52833DefaultPeripherals>,
+        nrf52833::chip::NRF52::new(nrf52833_peripherals)
+    );
+    CHIP = Some(chip);
+
+    // Need to disable the MPU because the bootloader seems to set it up.
+    chip.mpu().clear_mpu();
```

Comment:
Is this true on this board? It would be good to double check now (if you haven't) so we don't copy this line from the nano33 unnecessarily.

Reply:
I don't know, let me check.

Reply:
Deleted